griptape/drivers/vector/qdrant_vector_store_driver.py

```diff
@@ -0,0 +1,259 @@
+                port=self.port,
+                prefer_grpc=self.prefer_grpc,
+                grpc_port=self.grpc_port,
+            )
+        # Ensure the collection exists and has the correct configuration
+        self._create_collection(self.embedding_driver)
+
+    def delete_vector(self, vector_id: str) -> None:
+        """
+        Delete vectors from the Qdrant collection based on their IDs.
+
+        Parameters:
+        - vector_ids: Optional list of vector IDs to delete.
+        """
+        deletion_response = self.client.delete(
+            collection_name=self.collection_name, points_selector=rest.PointIdsList([vector_id])
+        )
+        if deletion_response.status == rest.UpdateStatus.COMPLETED:
+            return f"Ids {vector_id} is successfully deleted"
+
+    def query(
+        self, query: str, count: Optional[int] = None, namespace: Optional[str] = None, include_vectors: bool = False
+    ) -> list[BaseVectorStoreDriver.QueryResult]:
+        """
+        Query the Qdrant collection based on a query vector.
+
+        Parameters:
+        - query: Query string.
+        - count: Optional number of results to return.
+        - include_vectors: Whether to include vectors in the results.
+
+        Returns:
+        - List of QueryResult objects.
+        """
+
+        query_vector = self.embedding_driver.try_embed_chunk(query)
+
+        # Create a search request
+        limit = count
+        with_payload = True
+        with_vectors = True  # Ensure we get payloads in the results
+
+        results = self.client.search(
+            collection_name=self.collection_name,
+            query_vector=query_vector,
+            limit=limit,
+            with_payload=with_payload,
+            with_vectors=with_vectors,
+        )
+
+        # Convert results to QueryResult objects
+        query_results = [
+            BaseVectorStoreDriver.QueryResult(
+                id=result.id,
+                vector=result.vector if include_vectors else [],
+                score=result.score,
+                meta={k: v for k, v in result.payload.items() if k not in ["_score", "_tensor_facets"]},
+            )
+            for result in results
+        ]
+        # print(query_results)
+        return query_results
+
+    def upsert_vector(
+        self,
+        vector: list[float],
+        vector_id: Optional[str] = None,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        content: Optional[str] = None,
+    ) -> str:
+        """
+        Upsert vectors into the Qdrant collection.
+
+        Parameters:
+        - vector: The vector to be upserted.
+        - vector_id: Optional vector ID.
+        - namespace: Optional namespace for the vector.
+        - meta: Optional dictionary containing metadata.
+        - content: The text content to be included in the payload.
+        """
+
+        if vector_id is None:
+            vector_id = str(uuid.uuid4())
```

Comment:
Generate vector_id like this:
https://github.com/griptape-ai/griptape/blob/81194f1285192b8addb1cae92ac5e0b83130f770/griptape/drivers/vector/local_vector_store_driver.py#L22